vision/standard_object/odlc_classify_shape.py

```diff
@@ -69,420 +89,339 @@ def process_shapes(
-    )
+    polar: NDArray[Shape["*, 2"], Float64] = []
+    # Stores an array of angles and radii as tuples (radius, angle)
+    for i in range(len(arr)):
+        coord: Tuple[float, float] = cartesian_to_polar(arr[i][0][1], arr[i][0][0])
+        polar.append(coord)
+    return polar
 
 
-def classify_circular(contour: consts.Contour) -> chars.ODLCShape:
+def verify_shape_choice(mystery_radii_list: List[float], sample_ODLC_radii: List[float]) -> bool:
     """
-    Determines which circular shape the given contour is (assumed to be a circular shape)
+    Verifies that an ODLC's Polar graph is adequately similar to the "guessed" ODLC's sample graph
 
     Parameters
     ----------
-    contour : consts.Contour
-        The original contour to evaluate
+    mystery_radii_list : List[float]
+        The list of radii of the ODLC shape we are trying to classify
+    sample_ODLC_radii : List[float]
+        The sample list of radii for the guessed ODLC shape
 
     Returns
     -------
-    circular_shape : chars.ODLCShape
-        Will return one of CIRCLE, SEMICIRCLE, QUARTER_CIRCLE from ODLCShape Enum
-
-    Notes
-    -----
-    Alternative method to determine which circular shape could be to see how many right angles
-    the given shape has a quarter-circle may have 1 or 3 (if corner between straight line and
-    curve show up as a right angle), and a semicircle would have 0 or 2 for same reason. But
-    a regular circle would have 0 right angles, which may conflict with semicircle.
+    shape : bool
+        Returns true if the absolute difference of the two radii lists is small enough, false if not
     """
-    approx: consts.Contour = cv2.approxPolyDP(contour, cv2.arcLength(contour, True) * 0.01, True)
-    max_dist: npt.Float64 = np.max(get_lengths(approx))
-    perimeter: npt.Float64 = cv2.arcLength(contour, True)
-
-    if abs((max_dist / perimeter) - QUARTER_CIRCLE_RATIO) < CIRCULAR_RATIO_RANGE:
-        return chars.ODLCShape.QUARTER_CIRCLE
-
-    if abs((max_dist / perimeter) - SEMICIRCLE_RATIO) < CIRCULAR_RATIO_RANGE:
-        return chars.ODLCShape.SEMICIRCLE
-    return chars.ODLCShape.CIRCLE
-
-
-if __name__ == "__main__":
-    # make some test shape
-    main_shape: consts.Contour = np.array(
-        [[[0, 0]], [[5, 5]], [[0, 10]], [[10, 10]], [[5, 5]], [[10, 0]]]
-    )
-    # make an approximation
-    main_approx: consts.Contour = cv2.approxPolyDP(main_shape, APPROX_CNT_EPSILON, True)
-
-    print(type(main_shape), main_shape.shape, main_shape)
-    print(type(main_approx), main_approx.shape, main_approx)
-    # run each individual internal testing function to check behavior
-    print(check_concavity(main_shape))
-    print(check_polygons(main_approx))
-    print(classify_circular(main_shape))
+    difference: float = 0.0
+    for i in range(NUM_STEPS):
+        difference += abs(mystery_radii_list[i] - sample_ODLC_radii[i])
+    return difference < NUM_STEPS / 8
```

Comment:
Where does this 8 come from? If this is just a number that you think works, then make it a constant at the top of the file.